app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -217,6 +213,12 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
       .setOnValueChanged { button, value -> button.showIfTrue(value.isNotNullOrEmpty()) }
       .hide()
 
+  fun addRedoButton(clickHandler: () -> Unit, visiblityHandler: (TaskButton, TaskData?) -> Unit) =
+    addButton(ButtonAction.REDO)
+      .setOnClickListener { clickHandler() }
+      .setOnValueChanged { button, value -> visiblityHandler(button, value) }
+      .hide()
```

Comment:
Let's move this method to `app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/polygon/DrawAreaTaskFragment.kt` and make it private as it is not used anywhere else.